src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs

```diff
@@ -46,24 +46,38 @@ public bool IsAppHostOrSingleFileOrNativeAot
         && !IsCurrentTestApplicationHostDotnetMuxer
         && !IsCurrentTestApplicationHostMonoMuxer;
 
+    public string GetCurrentTestApplicationFullPath()
+    {
+        string? moduleName = TryGetCurrentTestApplicationFullPath();
+
+        ApplicationStateGuard.Ensure(moduleName is not null);
+        return moduleName;
+    }
+
 #if NETCOREAPP
     [UnconditionalSuppressMessage("SingleFile", "IL3000:Avoid accessing Assembly file path when publishing as a single file", Justification = "We handle the singlefile/native aot use case")]
 #endif
-    public string GetCurrentTestApplicationFullPath()
+    public string? TryGetCurrentTestApplicationFullPath()
     {
-#pragma warning disable IL3000
         // This is empty in native app, or in single file app.
         string? moduleName = Assembly.GetEntryAssembly()?.Location;
-#pragma warning restore IL3000
 
-        moduleName = RoslynString.IsNullOrEmpty(moduleName)
+        return RoslynString.IsNullOrEmpty(moduleName)
             ? GetProcessPath(_environment, _process)
             : moduleName;
+    }
 
-        ApplicationStateGuard.Ensure(moduleName is not null);
```

Comment:
This guard isn't correct. For example, in case of Wasm (which is obviously not supported currently), this guard fails.

Some callers don't really require non-null value. So, we now have `TryGetCurrentTestApplicationFullPath` (return is nullable) and `GetCurrentTestApplicationFullPath` (return is not nullable).